semaphoreci: update Debian release to bullseye

ddstreet opened the pull request:
The Debian upstream-ci branch is updating to require debhelper 13,
which is only currently available from bullseye, or buster-backports.

ddstreet commented:
apparently the entire ubuntu ci infrastructure is broken currently, it's timing out trying to connect to launchpad.net (even though it's running inside the same production network) :(

At least this allows semaphoreci to pass, and I don't think the ubuntu ci tests are relevant to this change.

evverx commented:
To judge from https://semaphoreci.com/systemd/systemd/branches/pull-request-16545/builds/1, the tests failed there with
```
test_network_manager (__main__.ServicesTest) ... Unit network-manager.service could not be found.
ERROR
...

======================================================================
ERROR: test_network_manager (__main__.ServicesTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.24s4dw0c/downtmp/build.ygk/src/debian/tests/boot-and-services", line 90, in test_network_manager
    self.active_unit('network-manager')
  File "/tmp/autopkgtest-lxc.24s4dw0c/downtmp/build.ygk/src/debian/tests/boot-and-services", line 181, in active_unit
    out = subprocess.check_output(['systemctl', 'status', unit])
  File "/usr/lib/python3.8/subprocess.py", line 411, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 512, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['systemctl', 'status', 'network-manager']' returned non-zero exit status 4.

----------------------------------------------------------------------
Ran 23 tests in 1.513s

FAILED (errors=1, skipped=7)
```
I've just restarted it to figure out whether it's a glitch or not

evverx commented:
Looks like `network-manager.service` no longer ships there. For this to work it seems `network-manager` should be replaced with `NetworkManager` in https://salsa.debian.org/systemd-team/systemd/-/blob/upstream-ci/debian/tests/boot-and-services#L90

evverx commented:
I have to admit that I have no idea what that test is supposed to test. Maybe it could simply be dropped. Since @mbiebl seems to be a maintainer of the network-manager package I think it would make sense to ask @mbiebl how to fix that properly.

evverx commented:
It seems to have been fixed in the debian/master branch: https://salsa.debian.org/systemd-team/systemd/-/commit/f28333a0f1e99d24177305f9e9c10e9d87609c01. Maybe it's time to somehow sync the two branches.

ddstreet commented:
> To judge from https://semaphoreci.com/systemd/systemd/branches/pull-request-16545/builds/1, the tests failed there with

I can't seem to get the complete log from the semaphoreci page, the 'Download complete log' link just gets be the truncated log...is there a trick to getting it?

> It seems to have been fixed in the debian/master branch: https://salsa.debian.org/systemd-team/systemd/-/commit/f28333a0f1e99d24177305f9e9c10e9d87609c01. Maybe it's time to somehow sync the two branches.

ok, i'll do a test run locally with a rebased branch and push it if it looks good.

ddstreet commented:
> It seems to have been fixed in the debian/master branch: https://salsa.debian.org/systemd-team/systemd/-/commit/f28333a0f1e99d24177305f9e9c10e9d87609c01. Maybe it's time to somehow sync the two branches.

yep that seems to have fixed it, semaphoreci is passing now.  the upstream-ci branch is now rebased on debian/master (plus one patch to temporarily reduce the debhelper version to 11, to allow semaphoreci to pass for other PR, which i added after running the ci for this commit)

evverx commented:
> is there a trick to getting it?

Not that I know of. I just downloaded the truncated log and looked for the relevant part in hopes it wasn't truncated there.
